src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs

```diff
@@ -1075,11 +1075,10 @@ private static NamedTypeSymbol GetDelegateType(BoundExpression expr)
                 methods = constructedMethods.ToImmutableAndFree();
             }
 
-            if (methods.Length == 1 && !extensionMethodsOfSameViabilityAreAvailable)
+            if (methods.Length == 1 && methods[0].Arity == 0 && !extensionMethodsOfSameViabilityAreAvailable)
```

Comment:
In that case type inference and binding of the lambda against method parameters occurred earlier in the pipeline when the method was considered as a possible candidate.

---
In reply to: [67027973](https://github.com/dotnet/roslyn/pull/11981#discussion_r67027973) [](ancestors = 67027973,67027377,67019197)